src/app/loadout/item-utils.ts

```diff
@@ -33,3 +42,31 @@ export async function pickSubclass(filterItems: (item: DimItem) => boolean) {
     return item;
   } catch (e) {}
 }
+
+export function getSubclassPlugs(
+  defs: D2ManifestDefinitions,
+  subclass: ResolvedLoadoutItem | undefined
+) {
+  const plugs: PluggableInventoryItemDefinition[] = [];
+
+  if (subclass?.item.sockets?.categories) {
+    for (const category of subclass.item.sockets.categories) {
+      const showInitial =
+        !aspectSocketCategoryHashes.includes(category.category.hash) &&
+        !fragmentSocketCategoryHashes.includes(category.category.hash);
+      const sockets = getSocketsByIndexes(subclass.item.sockets, category.socketIndexes);
+
+      for (const socket of sockets) {
+        const override = subclass.loadoutItem.socketOverrides?.[socket.socketIndex];
+        const initial = getDefaultAbilityChoiceHash(socket);
+        const hash = override || (showInitial && initial);
+        const plug = hash && defs.InventoryItem.get(hash);
+        if (plug && isPluggableItem(plug)) {
+          plugs.push(plug);
+        }
+      }
+    }
+  }
+
+  return plugs;
+}
```

Comment:
Needed to move this due to cyclic imports. There are no changes, just a straight move.

Reply:
I wonder if this is the correct function to call here -- DIM fills in holes in ability sockets with the default ability hash in UI, but doesn't actually revert these sockets when applying the loadout. This is arguably surprising, but `implementsDimLoadout` should probably check instead which plugs loadout application will actually apply.

Reply:
Yeah I wondered that but in the case where someone has left their abilities (say grenade) as the default and expects the loadout to use that, then we should apply it IMO. 

Why aren't we applying it when applying the loadout? Since we indicate in the UX that it is selected I would argue we should be changing that behaviour.

Reply:
is this a breaking concern or is this PR worth merging for now as Better Than Before and fine tuning later?

Reply:
I think this is a case of "better than before", I would like to come to a consensus on how we should be handling "unselected" abilities in DIM loadouts and then sync things. I will raise a thread in slack.

Reply:
It appears that we are saving default abilities as socket overrides anyway so I think this might be a moot point. The following is the save call to the backend when making no modifications to the subclass from the loadout edit screen (haven't checked other entry points to saving loadouts).

<img width="1128" alt="Screenshot 2023-04-17 at 11 14 35 am" src="https://user-images.githubusercontent.com/7344652/232356093-3fc9da99-c535-4c75-ad55-a27296fc6ca3.png">